lib/util/sql/obfuscate.js

```diff
@@ -7,7 +7,7 @@
 
 module.exports = obfuscate
 
-var singleQuote = /'(?:[^']|'')*?(?:\\'.*|'(?!'))/
+var singleQuote = /'(?:''|[^'])*?(?:\\'.*|'(?!'))/
 var doubleQuote = /"(?:[^"]|"")*?(?:\\".*|"(?!"))/
```

Comment:
Repeating comments left in the issue -- 

I see in `'(?:''|[^'])*?(?:\\'.*|'(?!'))` you've swapped the two conditions so that (in "PCRE comment format") instead of 

```
    /
        ...
        (?:
            [^'] | ''
        )
        ...
    /x
```

we have 

```
    /
        ...
        (?:
            '' | [^'] 
        )
        ...
    /x
```

That seems reasonable and the group still appears to capture "every single character that **isn't a quote** or _is two single quote characters_". We also have additional confidence that current agent behavior is maintained because it also passes our tests.